sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java

```diff
@@ -774,46 +775,35 @@ public Predicate createUserFilter(CriteriaBuilder cb, CriteriaQuery cq, From<?,
 		Predicate filterResponsible = null;
 		Predicate filter = null;
 
-		if (!currentUser.hasAnyUserRole(
-			UserRole.NATIONAL_USER,
-			UserRole.POE_NATIONAL_USER,
-			UserRole.NATIONAL_CLINICIAN,
-			UserRole.NATIONAL_OBSERVER,
-			UserRole.REST_USER)) {
+		final JurisdictionLevel jurisdictionLevel = UserRole.getJurisdictionLevel(currentUser.getUserRoles());
+		if (jurisdictionLevel != JurisdictionLevel.NATION && !currentUser.hasAnyUserRole(UserRole.REST_USER)) {
 			// whoever created the case or is assigned to it is allowed to access it
 			filterResponsible = cb.equal(casePath.join(Case.REPORTING_USER, JoinType.LEFT), currentUser);
 			filterResponsible = cb.or(filterResponsible, cb.equal(casePath.join(Case.SURVEILLANCE_OFFICER, JoinType.LEFT), currentUser));
 			filterResponsible = cb.or(filterResponsible, cb.equal(casePath.join(Case.CASE_OFFICER, JoinType.LEFT), currentUser));
 
 			// allow case access based on user role
-			if (currentUser.hasAnyUserRole(
-				UserRole.SURVEILLANCE_SUPERVISOR,
-				UserRole.CONTACT_SUPERVISOR,
-				UserRole.CASE_SUPERVISOR,
-				UserRole.POE_SUPERVISOR,
-				UserRole.EVENT_OFFICER,
-				UserRole.STATE_OBSERVER) && currentUser.getRegion() != null) {
+			if (jurisdictionLevel == JurisdictionLevel.REGION && currentUser.getRegion() != null) {
```

Comment:
I'd suggest to use a switch statement instead. Also in the other createUserFilter methods.

Reply:
but there is also the checks for not null if i add the switch statement we'd still remain also with those if statements

Reply:
Yes, it's more about code logic and readability: The most important part is the jurisdiction and a switch statement would reflect that. The null check would be within the case block.